src/update-beacons.ts

```diff
@@ -204,27 +204,41 @@ export const updateBeacons = async (providerSponsorBeacons: ProviderSponsorBeaco
       continue;
     }
 
-    // Check signed data is newer than on chain value
-    const isDataFresh = checkSignedDataFreshness(onChainData.data, newBeaconResponse);
-    if (!isDataFresh) {
-      logger.log(`Sign data older then on chain record. Skipping.`);
+    // Check that signed data is newer than on chain value
+    const isSignedDataFresh = checkSignedDataFreshness(onChainData.data, newBeaconResponse);
+    if (!isSignedDataFresh) {
+      logger.log(`Signed data older than on chain record for beacon with ID ${beaconUpdateData.beaconId}. Skipping.`);
       continue;
     }
 
-    // Check beacon condition
-    const shouldUpdate = await checkUpdateCondition(
-      onChainData.data,
-      beaconUpdateData.deviationThreshold,
-      newBeaconValue
+    // Check that on chain data is newer than hearbeat interval
+    const isOnchainDataFresh = checkOnchainDataFreshness(
+      onChainData.data.timestamp,
+      beaconUpdateData.heartbeatInterval
     );
-    if (shouldUpdate === null) {
-      logger.log(`Unable to fetch current beacon value for beacon with ID ${beaconUpdateData.beaconId}.`);
-      // This can happen only if we reach the total timeout so it makes no sense to continue with the rest of the beacons
-      return;
-    }
-    if (!shouldUpdate) {
-      logger.log(`Deviation threshold not reached for beacon with ID ${beaconUpdateData.beaconId}. Skipping.`);
-      continue;
+    if (!isOnchainDataFresh) {
+      logger.log(
+        `On chain data timestamp older than heartbeat for beacon with ID ${beaconUpdateData.beaconId}. Updating without condition check.`
+      );
+    } else {
+      // Check beacon condition
+      const shouldUpdate = await checkUpdateCondition(
+        onChainData.data,
+        beaconUpdateData.deviationThreshold,
+        newBeaconValue
+      );
+      if (shouldUpdate) {
+        logger.log(`Deviation threshold reached for beacon with ID ${beaconUpdateData.beaconId}. Updating.`);
+      }
+      if (shouldUpdate === null) {
+        logger.log(`Unable to fetch current beacon value for beacon with ID ${beaconUpdateData.beaconId}.`);
+        // This can happen only if we reach the total timeout so it makes no sense to continue with the rest of the beacons
+        return;
+      }
+      if (!shouldUpdate) {
+        logger.log(`Deviation threshold not reached for beacon with ID ${beaconUpdateData.beaconId}. Skipping.`);
+        continue;
+      }
```

Comment:
We don't need to do both `if(shouldUpdate)` and `if(!shouldUpdate)`.
```suggestion
      if (shouldUpdate === null) {
        logger.log(`Unable to fetch current beacon value for beacon with ID ${beaconUpdateData.beaconId}.`);
        // This can happen only if we reach the total timeout so it makes no sense to continue with the rest of the beacons
        return;
      }
      if (!shouldUpdate) {
        logger.log(`Deviation threshold not reached for beacon with ID ${beaconUpdateData.beaconId}. Skipping.`);
        continue;
      }
      
      logger.log(`Deviation threshold reached for beacon with ID ${beaconUpdateData.beaconId}. Updating.`);
```